sql/core/src/main/scala/org/apache/spark/sql/SQLContext.scala

```diff
@@ -477,6 +477,20 @@ class SQLContext(@transient val sparkContext: SparkContext)
   }
 
   /**
+   * :: DeveloperApi ::
+   * Creates a [[DataFrame]] from an [[java.util.List]] containing [[Row]]s using the given schema.
+   * It is important to make sure that the structure of every [[Row]] of the provided List matches
+   * the provided schema. Otherwise, there will be runtime exception.
+   *
+   * @group dataframes
+   * @since 1.6.0
+   */
+  @DeveloperApi
+  def createDataFrame(rows: java.util.List[Row], schema: StructType): DataFrame = {
+    createDataFrame(sparkContext.parallelize(rows.asScala), schema)
```

Comment:
Should create a local DataFrame using `LocalRelation`. See https://github.com/holdenk/spark/blob/SPARK-10630-create-DataFrame-from-Java-List/sql/core/src/main/scala/org/apache/spark/sql/SQLContext.scala#L385.